cluster-autoscaler/cloudprovider/hetzner/hetzner_cloud_provider.go

```diff
@@ -250,61 +237,45 @@ func BuildHetzner(_ config.AutoscalingOptions, do cloudprovider.NodeGroupDiscove
 			region:             strings.ToLower(spec.region),
 			targetSize:         len(servers),
 			clusterUpdateMutex: &clusterUpdateLock,
+			placementGroup:     placementGroup,
 		}
+	}
 
-		// If a placement group was specified, check with the API to see if it exists
-		if manager.clusterConfig.IsUsingNewFormat {
-
-			placementGroupRef := manager.clusterConfig.NodeConfigs[spec.name].PlacementGroup
-
-			if placementGroupRef == "" {
-				continue
-			}
-
-			ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-			defer cancel()
-
-			placementGroup, _, err := manager.client.PlacementGroup.Get(ctx, placementGroupRef)
-
-			// Check if an error occurred
-			if err != nil {
-				if err == context.DeadlineExceeded {
-					klog.Fatalf("Timed out checking if placement group `%s` exists.", placementGroupRef)
-				} else {
-					klog.Fatalf("Failed to verify if placement group `%s` exists error: %v", placementGroupRef, err)
-				}
-			}
-
-			// If the placement group exists, add it to the node group config
-			if placementGroup != nil {
-				manager.nodeGroups[spec.name].placementGroup = placementGroup
-			} else {
-				klog.Fatalf("The requested placement group `%s` does not appear to exist.", placementGroupRef)
-			}
+	// Check if placement groups spanned over multiple node groups exceeds max placement group size
+	for pgName, totalMaxSize := range placementGroupTotals {
+		if totalMaxSize > maxPlacementGroupSize {
+			klog.Fatalf(
+				"Placement group %s exceeds max placement group size of %d, size %d",
+				pgName,
+				maxPlacementGroupSize,
+				totalMaxSize,
+			)
 		}
 	}
 
-	// Get placement groups with total maxSize over the maximum allowed
-	maxPlacementGroupSize := 10
+	return provider
+}
 
-	largePlacementGroups := getLargePlacementGroups(manager.nodeGroups, maxPlacementGroupSize)
+func getPlacementGroup(manager *hetznerManager, placementGroupRef string) *hcloud.PlacementGroup {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
 
-	// Fail if we have placement groups over the max size
-	if len(largePlacementGroups) > 0 {
+	placementGroup, _, err := manager.client.PlacementGroup.Get(ctx, placementGroupRef)
 
-		// Gather placement group names
-		var placementGroupIDs string
-		for i, placementGroupID := range largePlacementGroups {
-			if i > 0 {
-				placementGroupIDs += ", "
-			}
-			placementGroupIDs += strconv.FormatInt(placementGroupID, 10)
+	// Check if an error occurred
+	if err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			klog.Fatalf("Timed out checking if placement group `%s` exists.", placementGroupRef)
```

Comment:
In this method we could also return a proper error and then call `klog.Fatalf` at the callsite (which does not return an error itself).